x/leverage/keeper/reserves.go

```diff
@@ -0,0 +1,52 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/umee-network/umee/x/leverage/types"
+)
+
+// GetReserveAmount gets the amount reserved of a specified token. On invalid asset, the reserved amount is zero.
+func (k Keeper) GetReserveAmount(ctx sdk.Context, denom string) sdk.Int {
+	store := ctx.KVStore(k.storeKey)
+	key := types.CreateReserveAmountKey(denom)
+	amount := sdk.ZeroInt()
+	bz := store.Get(key)
+	if bz != nil {
```

Comment:
Curious, can we bubble up the error or do you think it's wise to panic here?

Reply:
The panics here are on two cases:
- negative reserve amount
- failed to unmarshal sdk.Int (but it wasn't empty/nil)

Since those two cases shouldn't occur at all and the rest of the function works without an error return at all, I think we can keep it as a panic. This resembles how `bankkeeper.GetBalance` works - no error return, potential panic on unmarshal fail.

Reply:
ACK